src/main/java/com/faforever/client/chat/AbstractChatTabController.java

```diff
@@ -130,24 +126,51 @@ public abstract class AbstractChatTabController implements Controller<Tab> {
   protected final I18n i18n;
   protected final NotificationService notificationService;
-  protected final ReportingService reportingService;
   protected final UiService uiService;
   protected final EventBus eventBus;
   protected final WebViewConfigurer webViewConfigurer;
-  protected final ChatUserService chatUserService;
   protected final EmoticonService emoticonService;
-  private final ImageUploadService imageUploadService;
-  private final CountryFlagService countryFlagService;
+  protected final CountryFlagService countryFlagService;
 
   /**
    * Messages that arrived before the web view was ready. Those are appended as soon as it is ready.
    */
-  private final List<ChatMessage> waitingMessages;
-  private final IntegerProperty unreadMessagesCount;
-  private final ChangeListener<Boolean> resetUnreadMessagesListener;
-  private final ChangeListener<Number> zoomChangeListener;
-  private final ChangeListener<Boolean> tabPaneFocusedListener;
-  private final ChangeListener<Boolean> stageFocusedListener;
+  private final List<ChatMessage> waitingMessages = new ArrayList<>();
+  private final IntegerProperty unreadMessagesCount = new SimpleIntegerProperty();
+
+  @SuppressWarnings("FieldCanBeLocal")
+  private final ChangeListener<Number> unreadMessageCountListener = (observable, oldValue, newValue) -> incrementUnreadMessageCount(newValue.intValue() - oldValue.intValue());
+  private final ChangeListener<Number> zoomChangeListener = (observable, oldValue, newValue) -> updateZoomPreferences(newValue);;
+  private final ChangeListener<Boolean> tabPaneFocusedListener = (focusedTabPane, oldTabPaneFocus, newTabPaneFocus) -> {
+    if (newTabPaneFocus) {
+      JavaFxUtil.runLater(() -> messageTextField().requestFocus());
+    }
+  };
+  @SuppressWarnings("FieldCanBeLocal")
+  private final ChangeListener<Boolean> selectedChangeListener = (observable, oldValue, newValue) -> {
+    if (newValue) {
+      // Since a tab is marked as "selected" before it's rendered, the text field can't be selected yet.
+      // So let's schedule the focus to be executed afterwards
+      JavaFxUtil.runLater(messageTextField()::requestFocus);
+    }
+  };
+  private final ChangeListener<Boolean> stageFocusedListener = (window, windowFocusOld, windowFocusNew) -> {
+    if (getRoot() != null
+        && getRoot().getTabPane() != null
+        && getRoot().getTabPane().isVisible()) {
+      JavaFxUtil.runLater(() -> messageTextField().requestFocus());
+    }
+  };
+  @SuppressWarnings("FieldCanBeLocal")
+  private final ChangeListener<TabPane> tabPaneChangeListener = (tabPane, oldTabPane, newTabPane) -> addTabListeners(newTabPane);
+  private final ChangeListener<State> webViewStatChangeListener = (observable, oldValue, newValue) -> sendWaitingMessagesIfLoaded(newValue);
+
+  private final ChangeListener<Boolean> resetUnreadMessagesListener = (observable, oldValue, newValue) -> {
+    if (hasFocus()) {
+      setUnread(false);
+    }
+  };
```

Comment:
This is odd style and hard to read. Why not make these functions and reference them with this::<listenerMethode> where used?

Reply:
Yes I intend to go back and make as many of them functions where I can to simplify the reading of it.

Although in many places we can't use the this:: syntax since the function would only take the new value.